### ISSUE TITLE
Broadlink Mini 5f36 & IR Learn Code 

### ISSUE THREAD

**Issue by sanchj** [lprhodes/homebridge-broadlink-rm#551]
The IR code learn does not work ... When you press the learn button in home app, the front led in rm Mini does not turn on. And nothing happen in homebridge. Wait for 10 seconds and no results. Does somebody has a Broadlink Mini 5f36 working fine ? It seems to be some protocol issues with this model ...

**Comment by SaschaBuettner** [comment]
Hi,

I also have issues with learning and the Broadlink RM Mini 3 with the detected device type code: "5f36".

I bought it recently on Amazon using this link: https://www.amazon.de/gp/product/B01G1PZUK2/ref=ppx_yo_dt_b_asin_title_o01_s00?ie=UTF8&th=1

**Comment by martinrehfeld** [comment]
Another +1 from me. I got the same device as @SaschaBuettner on my desk, it comes with

    Firmware: 44057
    Model: RM mini 3
    FCC ID: 2ACDZ-RMMINI3-RM

To add a few more observations, device discovery does work for me

    homebridge[23755]: [2/4/2020, 9:13:54 PM] [Broadlink RM] [INFO] Discovered Broadlink RM Mini 3 D (5f36) at 192.168.*.* (a7:df:24:7a:*:*)

When configuring codes manually, the `sendHex` messages appear in the log without error, but the LED on the RM never comes on. Also, when issuing "Learn", the LED stays off all the time. It then times out without data as described above.

The Python folks and, by extension, Home Assistant also seems to have trouble with this specific model, see this issue for HA which also cross-references issues in the `python-broadlink` library: https://github.com/home-assistant/home-assistant/issues/30215

People have also started to look into packet dumps, see: https://github.com/home-assistant/home-assistant/issues/23566#issuecomment-581666729

**Comment by nickollasaranha** [comment]
Is there a way to get the hexa from the Broadlink App? Because I can learn things over there... Just doesn't work with this plugin nor Broadlink Manager

**Comment by sanchj** [comment]
Does someone know hot to solve this problem ? Is there any progress that can make RM Mini 5f36 works fine ? Thanks

**Comment by josehipo** [comment]
> Does someone know hot to solve this problem ? Is there any progress that can make RM Mini 5f36 works fine ? Thanks

No, I tried everything with no luck... Finally I got a second hand 2737 and everything works great.

**Comment by sanchj** [comment]
It seems there are some updates in python code. Does anybody know if these updates can solve this issue?

**Comment by fizulnizam** [comment]
Any updates on this?

**Comment by tomlea** [comment]
So I looked at the changes made to support it in the Python code, and compared it to how it's working in JS land.

I edited `homebridge-broadlink-rm-pro/node_modules/kiwicam-broadlinkjs-rm/index.js` as follows:

```diff
   //Use different headers for rm4 devices
    this.request_header = (rm4DeviceTypes[parseInt(deviceType, 16)] || rm4PlusDeviceTypes[parseInt(deviceType, 16)]) ? new Buffer([0x04, 0x00]) : new Buffer([]);
--    this.code_sending_header = (rm4DeviceTypes[parseInt(deviceType, 16)] || rm4PlusDeviceTypes[parseInt(deviceType, 16)]) ? new Buffer([0xda, 0x00]) : new Buffer([]);
++    this.code_sending_header = (rm4DeviceTypes[parseInt(deviceType, 16)] || rm4PlusDeviceTypes[parseInt(deviceType, 16)]) ? new Buffer([0xd0, 0x00]) : new Buffer([]);
```

I'm now seeing hex codes when I do learning with my 5f36:
```log
10/22/2020, 18:17:09] [Broadlink RM] [INFO] Discovered Broadlink RM Mini 3 D (5f36) at XXXXXX (XXXXXX)
[10/22/2020, 18:17:45] [Broadlink RM] Learn Code (ready)
[10/22/2020, 18:17:49] [Broadlink RM] [RESULT] Learn Code (learned hex code: 26005000000129941313131313121313131213131312131313361436133713360133713371237131313121313131313121313133613131336133713371237131436130005160001304a13000d05000000000000)
[10/22/2020, 18:17:49] [Broadlink RM] [INFO] Learn Code (complete)
[10/22/2020, 18:17:49] [Broadlink RM] [INFO] Learn Code (stopped)
```

Digging a little deeper, this is actually just a revert of https://github.com/kiwi-cam/broadlinkjs-rm/commit/a12fd007be3c38caddae8998bacda75a43cd5997

If anyone else would like to take this further, feel free, but this fixes it for me.
